src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs

```diff
@@ -2675,6 +2675,9 @@ internal void Run(bool inputLoopIsNested)
                         _parent._isRunningPromptLoop = false;
                     }
                 }
+#if UNIX
```

Comment:
I'm not sure this is right, it appears from this PR an _extra_ new line would be produced when `exit` is used.

Reply:
to overcome this, we should rework the code so to don't even print the prompt when EOF happens

Reply:
I noticed that there is already code that does something similar for Ctrl+C at https://github.com/PowerShell/PowerShell/blob/master/src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs#L2555-L2561.  I wonder if this case should be handled there as well.